reference/conanfile/attributes.rst

```diff
@@ -217,138 +217,137 @@ independent in VS, we can just remove that setting field:
-        requires = "Pkg/0.1@user/channel"
-        default_options = "Pkg:pkg_option=value"
+  .. code-block:: python
 
-You can also specify default option values of the conditional required dependencies:
+      class OtherPkg(ConanFile):
+          requires = "MyPkg/0.1@user/channel"
+          default_options = {"MyPkg:shared": False}
 
-.. code-block:: python
+  Of course, this will work in the same way working with a *conanfile.txt*:
 
-    class OtherPkg(ConanFile):
-        default_options = "Pkg:pkg_option=value"
+  .. code-block:: text
 
-        def requirements(self):
-            if self.settings.os != "Windows":
-                self.requires("Pkg/0.1@user/channel")
+      [requires]
+      MyPkg/0.1@user/channel
 
-This will always work, on Windows the `default_options` for the `Pkg/0.1@user/channel` will be ignored, they will only be used on every other os.
+      [options]
+      MyPkg:shared=False
 
-If you need to dynamically set some dependency options, you could do:
+- Through :ref:`profiles<profiles>` it is also possible to define default values for the
+  options of a recipe, and would be used whenever that recipe is used:
 
-.. code-block:: python
+  .. code-block:: text
 
-    class OtherPkg(ConanFile):
-        requires = "Pkg/0.1@user/channel"
+      # file "myprofile"
+      # use it as $ conan install -pr=myprofile
+      [settings]
+      setting=value
 
-        def configure(self):
-            self.options["Pkg"].pkg_option = "value"
+      [options]
+      MyPkg:shared=False
 
-Option values can be given in command line, and they will have priority over the default values in the recipe:
+- Last way of defining values for options, with the highest priority over them all, is to pass
+  these values using the command argument :command:`-o` in the command line:
 
-.. code-block:: bash
+  .. code-block:: bash
 
-    $ conan install . -o Pkg:shared=True -o OtherPkg:option=value
+    $ conan install . -o MyPkg:shared=True -o OtherPkg:option=value
 
-You can also defined them in consumer ``conanfile.txt``, as described in :ref:`this section<options_txt>`
+Values for options can be also conditionally assigned (or even deleted) in the methods
+``configure()`` and ``config_options()``, the
+:ref:`corresponding section<method_configure_config_options>` has examples documenting these
+use cases. However, conditionally assigning values to options can have it drawbacks as it is
+explained in the :ref:`mastering section<conditional_settings_options_requirements>`.
 
-.. code-block:: text
+One important notice is how these options values are evaluated and how the different conditionals
+that we can implement in Python will behave. As seen before, values for options can be defined
+in Python code (assigning a dictionary to ``default_options``) or through strings (using a
+``conanfile.txt``, a profile file, or through the command line). In order to provide a
+consistent implementation take into account these considerations:
```

Comment:
These _rules_ are the written brief of these tables: https://github.com/conan-io/conan/issues/3620